packages/pwa-buildpack/lib/WebpackTools/PWADevServer.js

```diff
@@ -164,54 +164,41 @@ const PWADevServer = {
 
         if (graphqlPlayground) {
             const endpoint = '/graphql';
-
             const oldBefore = webpackDevServerOptions.before;
+
             webpackDevServerOptions.before = (app, server) => {
                 oldBefore(app, server);
                 let middleware;
+
+                const readQueryFile = async filename => {
+                    const query = await readFile(filename, 'utf8');
+                    const name = path.relative(context, filename);
+
+                    return { endpoint, name, query };
+                };
+
                 const gatheringQueryTabs = new Promise((resolve, reject) => {
                     const { compiler } = server.middleware.context;
-                    compiler.hooks.done.tap(
-                        'PWADevServer',
-                        async ({ stats }) => {
-                            /**
-                             * Stats in an array because we have 2 webpack child
-                             * compilations, 1 for client and other for service worker.
-                             */
```

Comment:
Is this still true? The argument to this function certainly wasn't an array; perhaps this is still the case in production, though?

Also, the argument to this function appears to be `Stats {}` itself. I don't see how we could have ever been destructuring `stats` from the argument in the first place.

Reply:
Nice catch @jimbo. We changed the [webpack config to only have 1 compilation](https://github.com/magento/pwa-studio/pull/2390/files#diff-1acbde853507ec5d5dac59ef5301a593R81) during the build process.

Reply:
Funny, the tests failed now. May be we were mocking stuff and using that to test the plugin. Hence it never failed when I changed the `webpack.config.js` in #2390.

Reply:
Fixed!